src/mcts/node.h

```diff
@@ -206,33 +206,42 @@ class Node {
   std::string DebugString() const;
 
  private:
-  // List of edges.
+  // Turns out the alignment and padding of the Node was costing more than 8
+  // bytes on modern compilers. Simply rearranging the members leads to
+  // significant savings on padding. As such, instead of being organized
+  // conceptually, the members are merely ordered from largest to smallest.
+
+  // TODO: shrink the padding on this somehow? It takes 16 bytes even though
+  // only 10 are real! Maybe even merge it into this class??
   EdgeList edges_;
-  // Index of this node is parent's edge list.
-  uint16_t index_;
+
+  // Pointer to a parent node. nullptr for the root.
+  Node* parent_ = nullptr;
+  // Pointer to a first child. nullptr for a leaf node.
+  std::unique_ptr<Node> child_;
+  // Pointer to a next sibling. nullptr if there are no further siblings.
+  std::unique_ptr<Node> sibling_;
+
   // Average value (from value head of neural network) of all visited nodes in
   // subtree. For terminal nodes, eval is stored. This is from the perspective
   // of the player who "just" moved to reach this position, rather than from the
   // perspective of the player-to-move for the position.
   float q_ = 0.0f;
+  // Sum of policy priors which have had at least one playout.
+  float visited_policy_ = 0.0f;
   // How many completed visits this node had.
   uint32_t n_ = 0;
-  // (aka virtual loss). How many threads currently process this node (started
+
+  // Index of this node is parent's edge list.
+  uint16_t index_;
+  // (AKA virtual loss.) How many threads currently process this node (started
   // but not finished). This value is added to n during selection which node
   // to pick in MCTS, and also when selecting the best move.
   uint16_t n_in_flight_ = 0;
-  // Sum of policy priors which have had at least one playout.
-  float visited_policy_ = 0.0f;
 
-  // Does this node end game (with a winning of either sides or draw).
+  // Whether or not this node end game (with a winning of either sides or draw).
   bool is_terminal_ = false;
 
-  // Pointer to a parent node. nullptr for the root.
-  Node* parent_ = nullptr;
-  // Pointer to a first child. nullptr for a leaf node.
-  std::unique_ptr<Node> child_;
-  // Pointer to a next sibling. nullptr if there are no further siblings.
-  std::unique_ptr<Node> sibling_;
 
   // TODO(mooskagh) Unfriend NodeTree.
   friend class NodeTree;
```

Comment:
I think it would also be useful to add static assert just to watch total structure size and avoid surprises with other compilers, something like:
```cpp
// This assert is here just to track structure size. Feel free to change when changing Node object.
static_assert(sizeof(Node) == 52, "Unexpected size of Node");
```
right after Node structure.

Reply:
Done